plugin/src/shared/sc/plugin2021/util/GameRuleLogic.kt

```diff
@@ -60,7 +60,7 @@ object GameRuleLogic {
             if (throws) {
                 throw InvalidMoveException("Expected move from ${gameState.currentColor}", move)
             } else {
-                return MoveMistake.WRONG_COLOR
+                return WrongColor(move.color)
             }
```

Comment:
I dislike this duplication everywhere. If throwing really needs to be a feature (why?) the InvalidMoveException should simply take a MoveMistake rather than a custom string.

Reply:
yes, that's what I'm aiming for in the long run.
But I didn't want to rewrite everything in one go

Reply:
Now InvalidMoveExceptions are constructed from MoveMistakes which I turned back into an enum.